### PR TITLE
Nazar/feature/category button

### PR DESCRIPTION
**Pull request by Nazar-kob**

### CODE REVIEW

File: src/components/buttons/СategoryButton/CategoryButton.constans.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import Vegetable from '@assets/icons/category/vegetable.svg';
+import Fruit from '@assets/icons/category/fruit.svg';
+import Garden from '@assets/icons/category/garden.svg';
+
+export enum Categorys {
+	room = 'rooms',
+	garden = 'garden',
+	vegetable = 'vegetables',
+	fruit = 'fruits',
+}
```
Comment (Taraslysak) [inline src/components/buttons/СategoryButton/CategoryButton.constans.tsx:11]: IMHO, these kinds of enums should come from backend

Comment (Taraslysak) [reply src/components/buttons/СategoryButton/CategoryButton.constans.tsx:11]: Are we going to have only 4 options of categories? 
Is there an option that we will have more than 4 categories on future?

File: src/i18n/en.json [hunk 2]
```diff
@@ -6,6 +6,12 @@
 		"linkText": "Log in as a guest"
 	},
 	"components": {
-		"authBotton": "Log in with"
+		"authBotton": "Log in with",
```
Comment (Taraslysak) [inline src/i18n/en.json:9]: I still don't like nested dictionaries

File: src/components/buttons/СategoryButton/CategoryButton.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { sizes } from '@src/styling/sizes';
+
+type CategoryButtonProps = {
+	category: Categorys;
+	SvgIcon: React.FC<SvgProps>;
+	onPress: (category: Categorys) => void;
+};
+
+export const CategoryButton = ({ category, SvgIcon, onPress }: CategoryButtonProps) => {
+	const [isActive, setIsActive] = useState<boolean>(false);
```
Comment (Taraslysak) [inline src/components/buttons/СategoryButton/CategoryButton.tsx:18]: This is a state duplication. CategoryButton should not keep it's own "isActive' state, but rather receive it as Prop. We can safely assume that parent component of `CategoryButton` will have some array-like state of `selectedButtons` so the parent should decide if this button is active or not

File: src/components/buttons/СategoryButton/CategoryButton.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	const { t } = useTranslation();
+	const { styles } = useStyles(styleSheet);
+
+	const handleOnPres = () => {
+		setIsActive(!isActive);
+		onPress(category);
```
Comment (Taraslysak) [inline src/components/buttons/СategoryButton/CategoryButton.tsx:24]: This code only proves my theory. `const [isActive, setIsActive] = useState<boolean>(false);` is a. redundant state

File: src/styling/themes.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ export const lightTheme = {
 	colors: {
 		...DefaultTheme.colors,
 		borderGrey: '#0000002E',
+		borderGreen: '#325637',
```
Comment (Taraslysak) [inline src/styling/themes.ts:10]: I suggest we ask Ivanna if its possible for border color to be the same as primary color. I don't see the point in having 50 shades of green in project

**Review by Taraslysak — COMMENTED** [review]
Please consider some refactoring

File: src/components/buttons/СategoryButton/CategoryButton.constans.tsx (hunk 1 above)
Comment (Nazar-kob) [reply src/components/buttons/СategoryButton/CategoryButton.constans.tsx:11]: Right now we're only looking at four categories, So it doesn't make sense to pull categories from the backend.

Comment (nixissmtp) [reply src/components/buttons/СategoryButton/CategoryButton.constans.tsx:11]: Agree. Use python enum on backend to define data structure

**Review by nixissmtp — CHANGES_REQUESTED** [review]

**Pull request merged**